src/test/java/com/opentable/kafka/EmbeddedZookeeperTest.java

```diff
@@ -52,7 +52,8 @@ public void testBoots() throws Exception {
         final String connect = zk.getConnectString();
         ZooKeeper client = new ZooKeeper(connect, 60000, this);
 
-        while (!client.getState().isConnected()) {
+        final long since = System.currentTimeMillis();
+        while (events.isEmpty() && System.currentTimeMillis() - since < 30_000) {
```

Comment:
I know this is pretty minimal test code, but I've been enjoying a countdown latch, which handles the spurious wakeups for you.

Reply:
Yeah, I agree.  But I'm going to keep the scope limited right now.  We can replace that later for sure, especially if we add more tests that use this.

Reply:
Yup.  Sounds good.